connections.md

```diff
@@ -1,8 +1,252 @@
+
+TODO: Clarify that provisioning is only needed if not using an already provisioned DID.
+
+The _invitee_ will provision a new DID according to the DID method spec. For a Peer DID, this involves creating a matching peer DID and key. The newly provisioned DID and DID Doc is presented in the exchange_request message as follows:
+
+TODO: Add clear details about signing the did_doc attachment.
+
+##### Example
+
+```json
+{
+  "@id": "5678876542345",
+  "@type": "https://didcomm.org/didexchange/1.0/request",
+  "~thread": { "pthid": "<id of invitation>" },
+  "label": "Bob",
+  "did": "B.did@B:A",
+  "did_doc~attach": {
+    "mime-type": "",
+    "data": {
+      "base64": "<b64 encoded bytes of diddoc>"// DID Doc contents here.
+    }
+    "sig": <sig info>
+  }
+}
+```
+
+##### Attributes
+
+* The `@type` attribute is a required string value that denotes that the received message is an exchange request.
+* The [`~thread`](../../concepts/0008-message-id-and-threading/README.md#thread-object) decorator MUST be included:
+  * It MUST include the ID of the parent thread (`pthid`) such that the `request` can be correlated to the corresponding `invitation`. More on correlation [below](#correlating-requests-to-invitations).
+  * It MAY include the `thid` property. In doing so, implementations MUST set its value to that of `@id` on the same request message. In other words, the values of `@id` and `~thread.thid` MUST be equal.
+* The `label` attribute provides a suggested label for the DID being exchanged. This allows the user to tell multiple exchange requests apart. This is not a trusted attribute.
+* The `connection` attribute contains the `did` and `did_doc` attributes. This format maintains consistency with the Response message where this attribute is signed.
+* The `did` indicates the DID being exchanged.
+* The `did_doc~attach`  attachment is required when `did` contains a peer did. contains the DID Doc associated with the DID. If the DID method for the presented DID is not a peer method and the DID Doc is resolvable on a ledger, the `did_doc` attribute is optional. The attachment must include a signature
+
+##### Correlating requests to invitations
+
+An invitation is presented in one of two forms:
+
+* An explicit OutOfBand invitation with its own `@id`.
+* An [implicit](#implicit-invitation) invitation contained in a DID document's [`service`](https://w3c-ccg.github.io/did-spec/#service-endpoints) attribute.
+
+When a `request` responds to an explicit invitation, its `~thread.pthid` MUST be equal to the `@id` property of the invitation.
+
+When a `request` responds to an implicit invitation, its `~thread.pthid` MUST contain a [DID URL](https://w3c-ccg.github.io/did-spec/#dfn-did-url) that resolves to the specific `service` on a DID document that contains the invitation.
+
+TODO: Include examples using a ledger resolvable DID.
```

Comment:
wondering if we will be needing lots of real world examples, if we should choose to use `did:web` or something like that so its easy for us to update...